lldb/examples/python/crashlog.py

```diff
@@ -555,30 +555,44 @@ def load_images(self, options, loaded_images=None):
 
         futures = []
         with tempfile.TemporaryDirectory() as obj_dir:
-            with concurrent.futures.ThreadPoolExecutor() as executor:
 
-                def add_module(image, target, obj_dir):
-                    return image, image.add_module(target, obj_dir)
+            def add_module(image, target, obj_dir):
+                return image, image.add_module(target, obj_dir)
+
+            if options.parallel_image_loading:
+                with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
Instead of duplicating the code, I would limit the number of workers in the ThreadPool to one.

Reply:
Good point!